src/avram/criteria.cr

```diff
@@ -135,6 +135,11 @@ class Avram::Criteria(T, V)
     rows.query.group_by(column)
   end
 
+  # :nodoc:
+  def __reset_where : Avram::QueryBuilder
```

Comment:
Is there a way to make this private?
It feels weird to have to use `__` in a lang that has real method protection.

Reply:
I agree, it is something I would do in JS. The question is, should it be private? Things like `Thread` for example, are considered private in Crystal, while they are accessible.

Reply:
It may be possible but would require reworking how Criteria and Queries work. This needs to be public so the Queryable can call it, but it shouldn't be used by the public. It can be and it'll work but it just reads strangely so adding the `__` hides it a bit more.

We also do this in `distinct_on` so that you do `MyQuery.new.distinct_on(&.name)` isntead of `MyQuery.new.name.distinct_on` which reads super weird

Reply:
Hmm I'm not suer what you mean. If marked as `private` I am fairly certain it will not work since we are calling it from another class

Reply:
No, you're right, there isn't another way. You can't make it private. But I think the `__` can be omitted. Some methods and object in Crystal are considered private, while being publicly accessible. A mention in the docs would be sufficient, I think.